python/lsst/ts/genericcamera/driver/alliedvisioncamera.py

```diff
@@ -0,0 +1,302 @@
+
+        self.id = 0
+        self.camera = None
+        self.frame = None
+        self.exposure_time = None
+        self.is_live_exposure = False
+        self.liveview_use_autoexposure = True
+        self.normal_image_type = None
+        self.loop = asyncio.get_running_loop()
+        self.executor = concurrent.futures.ThreadPoolExecutor()
+
+        self.tags = FitsHeaderItemsGenerator().generate_fits_header_items(
+            FitsHeaderTemplate.STARTRACKER
+        )
+        self.log.debug(f"Tags: {self.tags}")
+
+    @staticmethod
+    def name():
+        """Set camera name."""
+        return "AlliedVision"
+
+    def initialise(self, config):
+        """Initialise the camera with the specified configuration file.
+
+        Parameters
+        ----------
+        config : str
+            The name of the configuration file to load."""
+        self.id = config.config["id"]
+        self.liveview_use_autoexposure = config.config["liveview_use_autoexposure"]
+        self.normal_image_type = getattr(vimba.PixelFormat, config.config["image_type"])
+
+        with vimba.Vimba.get_instance() as v:
+            self.camera = v.get_camera_by_id(self.id)
+            with self.camera:
+                # Try to adjust GeV packet size.
+                # This Feature is only available for GigE - Cameras.
+                try:
+                    self.camera.GVSPAdjustPacketSize.run()
+                    while not self.camera.GVSPAdjustPacketSize.is_done():
+                        pass
+                except (AttributeError, vimba.VimbaFeatureError):
+                    self.log.warning(
+                        "Camera is not GigE or packet size adjustment failed."
+                    )
+                # Try to enable ChunkMode
+                try:
+                    self.camera.ChunkModeActive.set(True)
+                except (AttributeError, vimba.VimbaFeatureError):
+                    self.log.warning("ChunkMode not available.")
+
+    def get_config_schema(self):
+        return yaml.safe_load(
+            """
+$schema: http://json-schema.org/draft-07/schema#
+description: Schema for Allied Vision cameras.
+type: object
+properties:
+  id:
+    default: DEV
+    type: string
+    description: The ID of the camera to be set in the FITS header.
+  liveview_use_autoexposure:
+    default: true
+    type: boolean
+    description: Flag to set if live view uses autoexposure or exposure time from takeImages.
+  image_type:
+    type: string
+    description: >
+      The image type to store. This usually provides information about the
+      pixel depth.
+"""
+        )
+
+    def get_make_and_model(self):
+        """Get the make and model of the camera.
+
+        Returns
+        -------
+        str
+            The make and model of the camera."""
+        return self.camera.get_name()
+
+    def get_roi(self):
+        """Gets the region of interest.
+
+        Returns
+        -------
+        int
```

Comment:
You are returning named variables, so you could probably do:

```
top: `int`
```

here and in the other cases.

Reply:
Fixed them all.